src/Test/EntityManagerMockFactoryTrait.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\Doctrine\Test;
+
+use Doctrine\Common\Persistence\Mapping\ClassMetadata;
+use Doctrine\ORM\AbstractQuery;
+use Doctrine\ORM\EntityManager;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\EntityRepository;
+use Doctrine\ORM\QueryBuilder;
+use PHPUnit\Framework\MockObject\MockObject;
+
+trait EntityManagerMockFactoryTrait
+{
+    /**
+     * @return EntityManagerInterface|MockObject
+     */
+    public function createEntityManagerMock(\Closure $qbCallback, $fields)
```

Comment:
This is an instance method, not a static method. 

Maybe this should be protected to reduce wrong usage...

Reply:
my bad i didn't that the method changed. it's better ! thanks. could you merge this PR ?